sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -428,6 +428,33 @@ object DataType {
     }
   }
 
+  /**
+   * Returns true if the two data types share the same shape by name, i.e. the names
```

Comment:
This looks a bit opposed to `equalsStructurally` in the definition of "shape". Maybe just say "the two data types have same fields and order recursively...".

Reply:
Sounds good, fine with the name of the function?

Reply:
I thought it might be a bit confusing with `equalsStructurally`, but I don't have better idea. As there is doc, should be fine.